Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -140,6 +148,26 @@ private void OnPostFlash(EntityUid uid, HeadRevolutionaryComponent comp, ref Aft
             return;
         }
 
+        // Check if the user has a ban on "Revolutionary"
+        if (mind != null && mind.Session != null)
+        {
+            var antagBans = _banManager.GetAntagBans(mind.Session.UserId);
+            var antagAllSelection = Loc.GetString("ban-panel-role-selection-antag-all-option");
+            var revolutionaryRoleId = "Rev"; // Assuming "Rev" is the ID for RevolutionaryRoleComponent
+
+            if (antagBans != null && (antagBans.Contains(antagAllSelection) || antagBans.Contains(revolutionaryRoleId)))
+            {
```

Comment:
Having to check all of this manually is very annoying instead of just being 1-2 method calls.

Reply:
Moved them all in to a common method IsAntagBanned on the ban manager for all 3 Revs, Zombie and antagSelectionSystem.